### DISCUSSION TITLE
INAV 6.0 FC lockup when using CMS

### DISCUSSION THREAD

**Discussion by breadoven** [iNavFlight/inav#8854] [General]
Just wondering if anyone else has had a problem using the CMS menu in 6.0 resulting in the FC locking up.

This is using a custom version of 6.0 so could be caused by custom changes although nothing of note has been changed affecting the CMS menu.

Happened 2 times in succession when testing on a multirotor. Occurred after landing with disarm on landing detection. Unfortunately it didn't happen another time so slightly random and quite possibly using the CMS could be a coincidence. There is a possibility it was related to going into the CMS from the Stats screen immediately after landing and disarming, but can't be sure. Happened in the Navigation menu trying to enter the RTH menu and also selecting back from the RTH menu the other time. CMS seemed to work OK after a reboot before first arm. 

Could be a bit difficult to debug given the random nature, difficulty replicating it and also there's no log because it's disarmed (probably wouldn't help anyway) so really just interested to know if others have had the same problem.

**Comment by Jetrell** [comment]
I haven't yet... But I only change things occasionally in the CMS after landing.
It wouldn't be the first time I've found a setting or case, that causes an issue when used in a specific sequence.
If you want to leave your Diff and the board type. I can runs some test under similar conditions ?

**Comment by breadoven** [comment]
I tried to replicate this again a number of times but couldn't so not sure why it happened. The quad had a heavy landing recently so might need to check for loose/chaffed wires or something similar I guess. 

One thing I did notice in the code is the normal stick commands aren't disabled when CMS is active which doesn't seem right. You could be inadvertently activating them whilst playing with the CMS menus. May as well do a fix for this even though I don't think it's anything to do with the lockup issue.

The lockup happened after 2 RTH landings with auto disarm but not with a manual Emergency Landing with auto disarm although it's hard to see how differences between the 2 types of landing would have an affect on the CMS, or stick movements in general I guess. All a bit random ... the worst type of problem.

Diff attached:
[INAV_6.0.0_cli_5_Inch Quad_20230126_161706 - Copy.txt](https://github.com/iNavFlight/inav/files/10909007/INAV_6.0.0_cli_5_Inch.Quad_20230126_161706.-.Copy.txt)

**Comment by breadoven** [comment]
Well I was checking using the HITL sim with a plane and the FC locked up using the CMS ... which reminded me that this has happened several times before using the HITL sim. Unfortunately I can't remember if this happened previously using the CMS or was random. I put it down to a HITL/MSP problem ... but maybe it wasn't, maybe it was an INAV CMS problem. If it is CMS related then it could be difficult to work out what's going on. On the other hand at least it only occurs when disarmed and not flying so ultimately an irritation more than anything else.

**Reply by Jetrell** [reply]
>On the other hand at least it only occurs when disarmed and not flying.

That's a plus for sure.

I compared my settings with yours. But nothing really stood out.. I've run many tests. But haven't had it lock up.
I'll still keep checking each time I go out flying.

**Comment by OptimusTi** [comment]
There were recent changes with the CMS correct? Maybe it's a new bug?
